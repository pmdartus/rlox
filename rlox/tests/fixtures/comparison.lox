print "Equals (==)";
print 1 == 1;
print true == true;
print false == false;
print nil == nil;
print "foo" == "foo";
print 1 == 2;
print true == false;

print "Greater (>)";
print 2 > 1;
print 1 > 1;
print 0 > 1;

print "Greater or equals (>=)";
print 2 >= 1;
print 1 >= 1;
print 0 >= 1;

print "Less (<)";
print 2 < 1;
print 1 < 1;
print 0 < 1;

print "Less (<=)";
print 2 <= 1;
print 1 <= 1;
print 0 <= 1;