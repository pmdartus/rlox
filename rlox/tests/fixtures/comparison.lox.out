Equals (==)
true
true
true
true
true
false
false
Greater (>)
true
false
false
Greater or equals (>=)
true
true
false
Less (<)
false
false
true
Less (<=)
false
true
true
